packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -206,15 +213,57 @@ void diagnoseXcodeBuildFailure(XcodeBuildResult result) {
     printError("  open ios/Runner.xcodeproj");
     return;
   }
+  if (result.buildForDevice
+      && result.buildCommands != null
+      && result.buildCommands.contains('build')
+      && result.appDirectory != null) {
+    RunResult checkBuildSettings = await runAsync(
+      result.buildCommands..add('-showBuildSettings'),
+      workingDirectory: result.appDirectory,
+      allowReentrantFlutter: true
+    );
+    if (checkBuildSettings.exitCode == 0
+        && !checkBuildSettings.stdout?.contains('DEVELOPMENT_TEAM') == true) {
```

Comment:
This is definitely the case with recent releases of Xcode (> 8.1, IIRC) when using automatic signing -- is it also the case in Xcode 7 and/or when you use manual mode and use a provisioning profile? 

One option would be to bump the minimum required Xcode version from its current value (IIRC, >= 7.0) in `mac.dart`.